nix/packages/checks/haskell-project.nix

```
@@ -0,0 +1,25 @@
+{
+  mkCheck,
+  ghciwatch,
+}:
+mkCheck {
+  name = "haskell-project";
+  sourceRoot = "source/tests/data/simple";
+  nativeBuildInputs = ghciwatch.haskellInputs;
+  inherit (ghciwatch) GHC_VERSIONS;
+
+  checkPhase = ''
+    # Need an empty `.cabal/config` or `cabal` errors trying to use the network.
+    mkdir .cabal
+    touch .cabal/config
+    export HOME=$(pwd)
```

This is a very minor thing, because it's functionally the same (Nix sets the initial `PWD` and `TMPDIR` to the root directory of the build), but `TMPDIR` captures the intention more accurately (we're using `PWD` as a temporary directory for `cabal`)

```suggestion
    export HOME="$TMPDIR"
```